tabpy-server/tabpy_server/app/app.py

```diff
@@ -285,48 +287,18 @@ def _validate_cert_key_state(msg, cert_valid, key_valid):
             log_and_raise(err, RuntimeError)
 
     def _parse_pwd_file(self):
+        succeeded = True
         if ConfigParameters.TABPY_PWD_FILE not in self.settings:
             logger.info("Password file is not specified")
         else:
-            pwd_file_name = self.settings[ConfigParameters.TABPY_PWD_FILE]
-            logger.info('Parsing password file %s' % pwd_file_name)
-
-            if not os.path.isfile(pwd_file_name):
-                logger.fatal('Password file not found')
-                return False
-
-            with open(pwd_file_name) as pwd_file:
-                pwd_file_reader = csv.reader(pwd_file, delimiter=' ')
-                for row in pwd_file_reader:
-                    # skip empty lines
-                    if len(row) == 0:
-                        continue
-
-                    # skip commented lines
-                    if row[0][0] == '#':
-                        continue
-
-                    if len(row) != 2:
-                        logger.error(
-                            'Incorrect entry "{}" '
-                            'in password file'.format(row))
-                        return False
-
-                    login = row[0].lower()
-                    if login in self.credentials:
-                        logger.error(
-                            'Multiple entries for username {} '
-                            'in password file'.format(login))
-                        return False
-
-                    self.credentials[login] = row[1]
-                    logger.debug('Found username {}'.format(login))
-
-            if len(self.credentials) == 0:
+            succeeded, self.credentials = parse_pwd_file(
```

Comment:
is "succeeded," needed here?  (I'm asking - doesn't look like it would be since it would be true always)

Reply:
parsing may fail due to many reasons, e.g. duplicated usernames, missing password hash, etc. Also, it may succeed and return empty credentials list for an empty file.